MIGRATION.md

```diff
@@ -2,6 +2,45 @@
 
 When new releases include breaking changes or deprecations, this document describes how to migrate.
 
+## Migrating to 0.14.0
+
+If migrating from below 0.13.17, you can run
+```
+dagster instance migrate
+```
+This optional migration makes performance improvements to the runs page in Dagit.
+
+### Breaking Changes
+- The Dagster Daemon now uses the same workspace.yaml file as Dagit to locate your Dagster code. You should ensure that if you make any changes to your workspace.yaml file, they are included in both Dagit’s copy and the Dagster Daemon’s copy. When you make changes to the workspace.yaml file, you don’t need to restart either Dagit or the Dagster Daemon - in Dagit, you can reload the workspace from the Workspace tab, and the Dagster Daemon will periodically check the workspace.yaml file for changes every 60 seconds. If you are using the Dagster Helm chart, no changes are required to include the workspace in the Dagster Daemon.
+- Dagster’s metadata API has undergone a signficant overhaul. Changes include:
+    - To reflect the fact that metadata can be specified on definitions in addition to events, the following names are changing. The old names are deprecated, and will function as aliases for the new names until 0.15.0:
+        - `EventMetadata` > `MetadataValue`
+        - `EventMetadataEntry` > `MetadataEntry`
+        - `XMetadataEntryData` > `XMetadataValue` (e.g. `TextMetadataEntryData` > `TextMetadataValue`)
+    - The `metadata_entries` keyword argument to events and Dagster types is deprecated. Instead, users should use the metadata keyword argument, which takes a dictionary mapping string labels to `MetadataValue`s.
+    - Arbitrary metadata on In/InputDefinition and Out/OutputDefinition is deprecated. In 0.15.0, metadata passed for these classes will need to be resolvable to `MetadataValue` (i.e. function like metadata everywhere else in Dagster).
+    - The description attribute of `EventMetadataEntry` is deprecated.
```

Comment:
I think it would be worthwhile to separate out the deprecations into their own heading, so that we don't risk readers thinking they're broken immediately.

Reply:
Thanks! I've moved the metadata deprecation into a separate block.